src/generator/state-machine-to-intermediate.js

```diff
@@ -225,6 +244,36 @@ export default ({timers, states}) => {
         };
     });
 
+    // Generate numeric labels for each label name
+    states.forEach(({state, statements}, index) => {
+        // Generate numeric label if string
+        if(typeof state === 'string') {
+            if(!labelMap[state]) {
+                const start = generateNextLabel(states);
+                labelMap[state] = start;
+                statements = statements.map((statement) => ({
+                    ...statement,
+                    start: statement.start + start,
+                }));
+            }
+            states[index].state = labelMap[state];
+            states[index].statements = statements;
+        }
+    });
+
+    // Update the goto with the new states
+    states.forEach(({statements}, stateIndex) => {
+        statements.forEach(({operations}, statementIndex) => {
+            operations.forEach((operation, operationIndex) => {
+                operation.forEach(({goto}, index) => {
+                    if(typeof goto === 'string') {
+                        states[stateIndex].statements[statementIndex].operations[operationIndex][index].goto = labelMap[goto];
```

Comment:
`operation.goto = labelMap[goto];` should work the same

Reply:
The foreach creates a copy of the operation, so if we update it, the original one is not updated

Reply:
It shouldn't do that, unless the code is using something like `Proxy`. `forEach` passes items to the callback by reference

Reply:
```suggestion
    for (const {statements} of states)
        for (const {operations} of statements)
            for (const operation of operations)
                for (const op of operation)
                    if(typeof op.goto === 'string')
                        op.goto = labelMap[op.goto];
```

Reply:
You are right, I just changed it, it was because of the foreach